autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -1457,3 +1469,167 @@ def _process_hyperparameters(self, hyperparameters, ag_args_fit=None, excluded_m
+                time_limits=None, hyperparameters=None, holdout_frac=None, verbosity=3,
+                models_name_suffix=None, teacher_preds='soft',
+                augmentation_data=None, augment_method='spunge', augment_args={'size_factor':5,'max_size':int(1e5)}):
+        """ Various distillation algorithms.
+            Args:
+                X,y: pd.DataFrame and pd.Series of training data.
+                    If None, original training data used during TabularPrediction.fit() will be loaded.
+                    This data is split into train/validation if X_test,y_test are None.
+                X_test,y_test: pd.DataFrame and pd.Series of validation data.
+                time_limits, hyperparameters, holdout_frac: defined as in TabularPrediction.fit()
+                teacher_preds (None or str): If None, we only train with original labels (no data augmentation, overrides augment_method)
+                    If 'hard', labels are hard teacher predictions given by: teacher.predict()
+                    If 'soft', labels are soft teacher predictions given by: teacher.predict_proba()
+                    Note: 'hard' and 'soft' are equivalent for regression problems.
+                    If augment_method specified, teacher predictions are only used to label augmented data (training data keeps original labels).
+                    To apply label-smoothing: teacher_preds='onehot' will use original training data labels converted to one-hots for multiclass (no data augmentation).  # TODO: expose smoothing-hyperparameter.
+                models_name_suffix (str): Suffix to append to each student model's name, new names will look like: 'MODELNAME_dstl_SUFFIX'
+                augmentation_data: pd.DataFrame of additional data to use as "augmented data" (does not contain labels).
+                    When specified, augment_method, augment_args are ignored, and this is the only augmented data that is used (teacher_preds cannot be None).
+                augment_method (None or str): specifies which augmentation strategy to utilize. Options: [None, 'spunge','munge']
+                    If None, no augmentation gets applied.
+                }
+                augment_args (dict): args passed into the augmentation function corresponding to augment_method.
+        """
+        hyperparameter_tune = False  # TODO: add as argument with scheduler options.
+        if augmentation_data is not None and teacher_preds is None:
+            raise ValueError("augmentation_data must be None if teacher_preds is None")
+
+        logger.log(20, f"Distilling with teacher_preds={str(teacher_preds)}, augment_method={str(augment_method)} ...")
+        if X is None:
+            X = self.load_X_train()
+            if not self.bagged_mode:
+                try:
+                    X_test = self.load_X_val()
+                    X_train = X
+                except FileNotFoundError:
+                    pass
+
+        if y is None:
```

Comment:
Added:
```
if X is None:
            if y is not None:
                raise ValueError("X cannot be None when y specified.")
```
to handle this issue.  However, I think it's generally useful to separate X,y loading since somebody may theoretically want to use different y's than the original data.